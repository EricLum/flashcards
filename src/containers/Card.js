import React, {Component} from 'react'
class Card extends Component {

  constructor(props) {
    super(props);
    this.toggleShowAnswer = this.toggleShowAnswer.bind(this)
  }

  state = {
    showAnswer: false
  }

  toggleShowAnswer(){
    this.setState(prevState => ({
      showAnswer: !prevState.showAnswer
    }))
  }

  render(){

    return(
      <div className='card'>
        <div>
          {this.props.title}
        </div>
        <div>
          {this.props.question}
        </div>
        <button onClick={this.toggleShowAnswer}>click me</button>
        {this.state.showAnswer &&
          <div>
            {this.props.answer}
          </div>
        }
      </div>
    )
  }
}

Card.defaultProps = {
  title: 'Javascript Question 1',
  question: 'Explain the concept of scope',
  answer: 'Scope is the realm in which a variable lives'
}

export default Card
