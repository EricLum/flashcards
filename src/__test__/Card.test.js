import React from 'react';
import Enzyme,{ shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import Card from '../containers/Card'

Enzyme.configure({adapter: new Adapter()})
 
describe('Card', () => {
  test('should be defined', () => {
    expect(Card).toBeDefined();
  });
  it('should render correctly', () => {
    const wrapper = shallow(<Card />);
    expect(wrapper).toMatchSnapshot()
  })
})

it('two plus two is four', () => {
  expect(2 + 2).toBe(4);
})

test('object assignment', () => {
  const data = {one: 1};
  data['two'] = 2;
  expect(data).toEqual({one: 1, two: 2})
})

test('adding positive numbers is not zero', () => {
  for(let a = 1; a < 10; a++){
    for (let b =1; b< 10; b++){
      expect (a+b).not.toBe(0);
    }
  }
})

test('there is no I in team', () => {
  expect('team').not.toMatch(/I/);
})

const ShoppingList = ['bananas', 'apples', 'pears']
test('the shopping list has apples', () => {
  expect(ShoppingList).toContain('apples');
})

function forEach(items, callback){
  for (let index=0; index< items.length; index++){
    callback(items[index]);
  }
}

const mockCallback = jest.fn();
forEach([0,1], mockCallback);
console.log(mockCallback.mock.calls)
expect (mockCallback.mock.calls.length).toBe(2);
