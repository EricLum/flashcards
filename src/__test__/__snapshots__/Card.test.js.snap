// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    answer="Scope is the realm in which a variable lives"
    question="Explain the concept of scope"
    title="Javascript Question 1"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          Javascript Question 1
</div>,
        <div>
          Explain the concept of scope
</div>,
        <button
          onClick={[Function]}
>
          click me
</button>,
        false,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Javascript Question 1",
        },
        "ref": null,
        "rendered": "Javascript Question 1",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Explain the concept of scope",
        },
        "ref": null,
        "rendered": "Explain the concept of scope",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "click me",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "click me",
        "type": "button",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            Javascript Question 1
</div>,
          <div>
            Explain the concept of scope
</div>,
          <button
            onClick={[Function]}
>
            click me
</button>,
          false,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Javascript Question 1",
          },
          "ref": null,
          "rendered": "Javascript Question 1",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Explain the concept of scope",
          },
          "ref": null,
          "rendered": "Explain the concept of scope",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "click me",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "click me",
          "type": "button",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
